Dart/src/com/jetbrains/lang/dart/ide/findUsages/DartServerFindUsagesHandler.java

```diff
@@ -64,6 +66,19 @@ public boolean processInReadAction(SearchResult result) {
     return true;
   }
 
+  private static PsiElement findPsiElement(PsiElement context, Location location) {
+    final String locationFilePath = location.getFile();
+    final PsiFile psiFile = findPsiFile(context, locationFilePath);
+    if (psiFile != null) {
+      final int offset = location.getOffset();
+      PsiElement result;
+      if ((result = PsiTreeUtil.findElementOfClassAtOffset(psiFile, offset, DartReference.class, false)) != null) {
+        return result;
+      }
+    }
+    return null;
```

Comment:
Good catch.
Actually, we don't need to search for any specific PsiElement, just and is OK.
We can then climb up to a class or unit members.

Ouch... Documentation comments are not part of elements they comment.
So, references in comments are all grouped in class or unit instead of methods/classes/etc.
![image](https://cloud.githubusercontent.com/assets/384794/9011586/feec3ef0-3762-11e5-9d7e-015f1481f2d1.png)